src/lib/core/WeaveTLVWriter.cpp

```diff
@@ -177,6 +178,43 @@ void TLVWriter::Init(uint8_t *buf, uint32_t maxLen)
     FinalizeBuffer = NULL;
 }
 
+/**
+ * Initializes a TLVWriter object to write into a dynamic buffer.
+ *
+ * @param[in]   buf     A pointer to the dynamic buffer into which TLV should be written.
+ * @param[in]   maxLen  The maximum number of bytes that should be written to the output buffer.
+ * @param[in]   allowResize
+ *                      If true, allocating new space as needed to store the data written.  If
+ *                      false, writing will fail with WEAVE_ERROR_BUFFER_TOO_SMALL if the written
+ *                      data exceeds the initial size of the dynamic buffer
+ */
+void TLVWriter::InitMalloced(uint8_t **&buf, uint32_t initialBufSize, uint32_t maxLen)
+{
+    buf = (uint8_t **)&mBufHandle;
+
+    if (initialBufSize <= 0)
+    {
+        initialBufSize = 1;
+    }
+    *buf = (uint8_t *) malloc(initialBufSize);
+
+    mBufStart = mWritePoint = *buf;
+    mMaxLen = maxLen;
+    mRemainingLen = initialBufSize;
+    if (mRemainingLen > mMaxLen)
+    {
+        mRemainingLen = mMaxLen;
+    }
```

Comment:
This, potentially, may not be required with some of the checks at the beginning.